u2f-ref-code/java/src/com/google/u2f/key/messages/TransferAccessResponse.java

```diff
@@ -0,0 +1,81 @@
+package com.google.u2f.key.messages;
+
+import java.util.Arrays;
+import java.util.Objects;
+
+import com.google.common.base.Preconditions;
+import com.google.u2f.server.ControlFlags;
+
+/**
+ * TransferAccessResponse is constructed by a device to which access has been transferred. It wraps 
+ * the TransferAccessMessage chain with necessary information to register a new key.
+ */
+public class TransferAccessResponse extends AuthenticateResponse {
+  private final TransferAccessMessage[] transferAccessMessages;
+  private final byte[] keyHandle;
+
+  /**
+   * Constructor for the TransferAccessResponse
+   * 
+   * @param controlFlags: Sets the {@link ControlFlags} for this Response.
+   * @param transferAccessMessages: Array of TransferAccessMessages. The first message is the first
+   *        in the chain.
+   * @param keyHandle: KeyHandle for the final key to be registered.
+   * @param counter: Integer
+   * @param signature: Signature using the attestation private key over ControlByte, Counter,
+   *        Challenge from the AuthenticateRequest, KeyHandle, and a hash of the last
+   *        TransferAccessMessage in the chain.
+   */
+  public TransferAccessResponse(byte controlFlags, TransferAccessMessage[] transferAccessMessages,
+      byte[] keyHandle, int counter, byte[] signature) {
+    super(controlFlags, counter, signature); 
+
+    Preconditions.checkNotNull(controlFlags, "Control flags should not be null");
+    Preconditions.checkNotNull(counter, "Counter should not be null");
```

Comment:
`controlFlags` and `counter` can't be null, they're primitive types.

Also, the non-null check for signature should be done in the parent class.